### ISSUE TITLE
Add a `TransactionResult` type.

### ISSUE THREAD

**Issue by jmalloc** [dogmatiq/infix#227]
The transaction result would be returned from `Transaction.Commit()` on success, and include the offset of the first event saved within this transaction, if any.

**Comment by jmalloc** [comment]
Rather than putting the offset in this new result type, we should place the same "item" slices that appear in `pipeline.Response`, that will help with #210 and eventually they can be removed from pipeline.Response.

**Comment by danilvpetrov** [comment]
So generally the Transaction interface inside `persistence` pkg would look like the following (including the new `TransactionResult` type):

```go
// Transaction exposes persistence operations that can be performed atomically.
// Transactions are not safe for concurrent use.
type Transaction interface {
	aggregatestore.Transaction
	eventstore.Transaction
	queuestore.Transaction
	offsetstore.Transaction

	// Commit applies the changes from the transaction.
	Commit(ctx context.Context) (TransactionResult, error)

	// Rollback aborts the transaction.
	Rollback() error
}

// TransactionResult is the result returned by Transaction.Commit() method.
type TransactionResult struct {
	EventItems []*eventstore.Item
	QueueItems []*queuestore.Item
}
```
Is this what you're thinking?

**Comment by jmalloc** [comment]
Yep, exactly!

**Comment by danilvpetrov** [comment]
Fixed by  #235.
